testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *	The function will compute the number of bytes in the string to which s points,
+ *	not including the terminating NUL character. and will return the length of s.
+ *
+ * method:
+ *	-Define two sample strings s1 and s2.
+ *	-Concatenate s1 and s2 using strcat and
+ *	store it in s3.
+ *	-Calculate number of bytes manually for s3.
+ *	-Get the string length using strlen() function for s3.
+ *	-Compare return value from strlen() and expceted one.
+ *
+*/
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#include "../testfrmw/testfrmw.h"
+#include "../testfrmw/testfrmw.c"
+
+#define TNAME "strlen/1-1.c"
+#define MAX_ARRAY_SIZE 15
+int main(void)
+{
+	int ret = 0, i = 0, actual_len = 0;
+	char sample_str_1[MAX_ARRAY_SIZE] = "HELLO ";
+	char sample_str_2[] = "WORLD";
+	char *sample_str_3 = NULL;
+
+	//Just to avoid compliler optimization, we are using strcat to get required string,
+	//instead of taking string directly.
+	sample_str_3 = strcat(sample_str_1, sample_str_2);
+
```

Comment:
This is a bit ugly.

Also this really lacks reasonable coverage for the strlen(). Why do we try to assert a single pre-defined string when we can easily generate random strings from zero length even up to several kilobytes in a loop and assert that what we got from strlen is exactly the length of the string we generated.